tests/common/pkg_formats/test_python.py

```diff
@@ -472,19 +474,26 @@ def test_basepydist_load_requires_provides_file():
 
 
 def test_dist_get_paths():
-    content = 'foo/bar,sha256=1,"45"\nfoo/spam,,\n'
-    temp_path, fpaths = _create_test_files((("", "SOURCES.txt", content),))
-
-    sp_dir = get_python_site_packages_short_path("2.7")
-
-    dist = PythonEggInfoDistribution(temp_path, "2.7", None)
-    output = dist.get_paths()
-    expected_output = [
-        (join_url(sp_dir, "foo", "bar"), "1", 45),
-        (join_url(sp_dir, "foo", "spam"), None, None),
-    ]
-    _print_output(output, expected_output)
-    assert output == expected_output
+    all_algos = hashlib.algorithms_guaranteed.union(
+        {alg.upper() for alg in hashlib.algorithms_guaranteed}
+    )
+    dummy_data = "This is a a string to be hashed for a unit test"
+    for algo in all_algos:
+        digest = hashlib.new(algo, dummy_data.encode("utf-8")).digest()
+        encoded_digest = urlsafe_b64encode(digest)
+        content = f'foo/bar,{algo}={encoded_digest},"45"\nfoo/spam,,\n'
+        temp_path, _ = _create_test_files((("", "SOURCES.txt", content),))
+
+        sp_dir = get_python_site_packages_short_path("3.10")
+
+        dist = PythonEggInfoDistribution(temp_path, "3.10", None)
+        output = dist.get_paths()
+        expected_output = [
+            (join_url(sp_dir, "foo", "bar"), encoded_digest, 45),
+            (join_url(sp_dir, "foo", "spam"), None, None),
+        ]
+        _print_output(output, expected_output)
+        assert output == expected_output
```

Comment:
@iamthebot,

This test is currently not passing. I have a suggested improvement for you:

```python
ALGORITHMS = hashlib.algorithms_guaranteed.difference({"md5", "sha1", "shake_128", "shake_256"})


@pytest.mark.parametrize("algorithm", ALGORITHMS)
def test_dist_get_paths(algorithm):
    dummy_data = "This is a a string to be hashed for a unit test"

    digest = hashlib.new(algorithm, dummy_data.encode("utf-8")).digest()
    encoded_digest = urlsafe_b64encode(digest)
    content = f'foo/bar,{algorithm}={encoded_digest},"45"\nfoo/spam,,\n'
    temp_path, _ = _create_test_files((("", "SOURCES.txt", content),))

    sp_dir = get_python_site_packages_short_path("3.10")

    dist = PythonEggInfoDistribution(temp_path, "3.10", None)
    output = dist.get_paths()
    expected_output = [
        (join_url(sp_dir, "foo", "bar"), str(encoded_digest), 45),
        (join_url(sp_dir, "foo", "spam"), None, None),
    ]
    _print_output(output, expected_output)
    assert output == expected_output
``` 

This version uses pytest parameters, which we prefer. It also leaves out the problematic algorithms.

Were you expecting this test to pass with upper case algorithm names? These upper case names are also causing this test to fail.